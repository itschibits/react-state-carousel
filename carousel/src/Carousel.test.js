import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";

it("renders without crashing", function() {
  render(<Carousel cardData='[
    {
      src: ./image1.jpg,
      caption: "Photo by Richard Pasquarella on Unsplash"
    },
    {
      src: ./image2.jpg,
      caption: "Photo by Pratik Patel on Unsplash"
    },
    {
      src: ./image3.jpg,
      caption: "Photo by Josh Post on Unsplash"
    }
  ]' title="Shells from far away beaches."/>)
});


it("works when you click on the right arrow", function() {
  const { container } = render(<Carousel />);

  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="Photo by Richard Pasquarella on Unsplash"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="Photo by Pratik Patel on Unsplash"]')
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = container.querySelector(".fa-chevron-circle-right");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  expect(
    container.querySelector('img[alt="Photo by Richard Pasquarella on Unsplash"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="Photo by Pratik Patel on Unsplash"]')
  ).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});

it("works when you click on the left arrow", function() {
  const { container } = render(<Carousel />);

  // goes forward one image to reveal left arrow
  const rightArrow = container.querySelector(".fa-chevron-circle-right");
  fireEvent.click(rightArrow);
  
  // move backward in the carousel
  const leftArrow = container.querySelector(".fa-chevron-circle-left");
  fireEvent.click(leftArrow);

  // expect the last image to show, but not the first
  expect(
    container.querySelector('img[alt="Photo by Richard Pasquarella on Unsplash"]')
    ).toBeInTheDocument();
    expect(
      container.querySelector('img[alt="Photo by Josh Post on Unsplash"]')
    ).not.toBeInTheDocument();
});

it("successfully hides left arrow at beginning of array", function() {
  const { container } = render(<Carousel />);
  expect(
    container.querySelector('.fa-chevron-circle-left')
  ).not.toBeInTheDocument();

});

it("successfully hides right arrow at end of array", function() {
  const { container } = render(<Carousel />)

  const rightArrow = container.querySelector(".fa-chevron-circle-right");
  fireEvent.click(rightArrow);
  fireEvent.click(rightArrow);

  expect(
    container.querySelector('.fa-chevron-circle-right')
  ).not.toBeInTheDocument();

});

// const { container } = render(<Carousel 
//   cardData={[
//     {
//       src: "./image1.jpg",
//       caption: "Photo by Richard Pasquarella on Unsplash"
//     }
//   ]} title="Shells from far away beaches." />)