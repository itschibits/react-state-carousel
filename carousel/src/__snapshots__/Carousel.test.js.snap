// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works when you click on the right arrow 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      Shells from far away beaches.
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="fas fa-chevron-circle-left fa-2x"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          Photo by Pratik Patel on Unsplash
        </h4>
        <img
          alt="Photo by Pratik Patel on Unsplash"
          class="Card-image"
          src="image2.jpg"
        />
        <small
          class="Card-small"
        >
          Image 
          2
           of 
          3
          .
        </small>
      </div>
      <i
        class="fas fa-chevron-circle-right fa-2x"
      />
    </div>
  </div>
</div>
`;
